clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1096,9 +1096,26 @@ Node selectReadReplica(TopicPartition partition, Node leaderReplica, long curren
     private Map<Node, FetchSessionHandler.FetchRequestData> prepareFetchRequests() {
         Map<Node, FetchSessionHandler.Builder> fetchable = new LinkedHashMap<>();
 
-        // Ensure the position has an up-to-date leader
-        subscriptions.assignedPartitions().forEach(
-            tp -> subscriptions.maybeValidatePositionForCurrentLeader(tp, metadata.currentLeader(tp)));
+        // Ensure the position has an up-to-date leader, if the leader is set and it's ApiVersion is new enough
+        subscriptions.assignedPartitions().forEach(tp -> {
+            Metadata.LeaderAndEpoch leaderAndEpoch = metadata.currentLeader(tp);
+            final boolean offsetForEpochAvailable;
```

Comment:
I think we can simplify this a little bit. The common case is that we will have an active connection and api version information available. If we are not connected, could we just go through `maybeValidatePositionForCurrentLeader`. as before?